packages/models-library/src/models_library/settings/base.py

```diff
@@ -0,0 +1,35 @@
+from contextlib import suppress
+from typing import List, Tuple
+
+from pydantic import BaseSettings, Extra, SecretStr, ValidationError
+
+
+class BaseCustomSettings(BaseSettings):
+    class Config:
+        case_sensitive = False
+        extra = Extra.forbid
+        allow_mutation = False
+        frozen = True
+        validate_all = True
+        json_encoders = {SecretStr: lambda v: v.get_secret_value()}
```

Comment:
what are these 2 for?
validate_all: will validate field defaults
json_encoders: does this mean this will automatically show the value in the secret?. ok that's interesting

Reply:
yes. 
the idea is to make sure the defaults introduced by the developer are validated as well. Notice that we have a special default factory in ``BaseCustomSettings`` that creates defaults of the sub-settings

Regarding the json_encoders, this means that passwords are visible in jsons. It is not so critical since json in settings are just used in the CLI. Anybody having access to the CLI has access to the environs already. In any case, if this is unsecure we can remove it. It is not that critical for the settings